search/controllers/classic/__init__.py

```diff
@@ -0,0 +1,174 @@
+        # Check arxiv id validity
+        for arxiv_id in id_list:
+            try:
+                parse_arxiv_id(arxiv_id)
+            except ValueError:
+                raise ValidationError(
+                    message="incorrect id format for {}".format(arxiv_id),
+                    link=("http://arxiv.org/api/errors#"
+                          "incorrect_id_format_for_{}").format(arxiv_id)
+                )
+    else:
+        id_list = None
+
+    # Parse result size.
+    try:
+        max_results = int(params.get("max_results", 50))
+    except ValueError:
+        raise ValidationError(
+            message="max_results must be an integer",
+            link="http://arxiv.org/api/errors#max_results_must_be_an_integer",
+        )
+    if max_results < 0:
+        raise ValidationError(
+            message="max_results must be non-negative",
+            link="http://arxiv.org/api/errors#max_results_must_be_non-negative"
+        )
+
+    # Parse result start point.
+    try:
+        start = int(params.get("start", 0))
+    except ValueError:
+        raise ValidationError(
+            message="start must be an integer",
+            link="http://arxiv.org/api/errors#start_must_be_an_integer"
+        )
+    if start < 0:
+        raise ValidationError(
+            message="start must be non-negative",
+            link="http://arxiv.org/api/errors#start_must_be_non-negative"
+        )
+
+    try:
+        query = ClassicAPIQuery(
+            phrase=phrase, id_list=id_list, size=max_results, page_start=start
+        )
+    except ValueError:
+        raise BadRequest(
+            "Either a search_query or id_list must be specified"
+            " for the classic API."
+        )
+
+    # pass to search indexer, which will handle parsing
+    document_set: DocumentSet = index.SearchSession.current_session().search(
+        query
+    )
+    data: SearchResponseData = {"results": document_set, "query": query}
+    logger.debug(
+        "Got document set with %i results", len(document_set["results"])
+    )
+
+    # bad mypy inference on TypedDict and the status code
+    return data, status.HTTP_200_OK, {}  # type:ignore
+
+
+def paper(paper_id: str) -> Tuple[Dict[str, Any], int, Dict[str, Any]]:
+    """
+    Handle a request for paper metadata from the API.
+
+    Parameters
+    ----------
+    paper_id : str
+        arXiv paper ID for the requested paper.
+
+    Returns
+    -------
+    dict
+        Response data (to serialize).
+    int
+        HTTP status code.
+    dict
+        Extra headers for the response.
+
+    Raises
+    ------
+    :class:`NotFound`
+        Raised when there is no document with the provided paper ID.
+
+    """
+    try:
+        document = index.SearchSession.current_session().get_document(
+            paper_id
+        )  # type: ignore
+    except index.DocumentNotFound as e:
+        logger.error("Document not found")
+        raise NotFound("No such document") from e
```

Comment:
```suggestion
        raise NotFound("No such document") from ex
```

Reply:
Do you generally prefer the exception variable to be called `ex`? I can change that everywhere?

Asking because in this case `ex` doesn't exist in this context.

Reply:
Our main preference is to have variable names that are at least two characters long, but we haven't been completely consistent about this either.

Reply:
OK will change it wherever I find one.

Reply:
Done.